### ISSUE TITLE
[deployer][engine] External commits do not trigger the context rebuilds

### ISSUE THREAD

**Issue by tfield** [craftercms/craftercms#4616]
## Describe the bug
This is a follow up to a bugfix I found in 3.1.11:

https://github.com/craftercms/craftercms/issues/4485

In 3.1.13, saving either one of the two application-context.xml files (and then committing it), or saving groovy classes (in /scripts/classes) does not trigger a rebuild of the application context, but they are required.

## To Reproduce
1. Create a groovy class & call it via a controller/rest service
2. Change the class via your IDE
3. Commit the change
4. Note changes not reflected - context has not reloaded

## Expected behavior
It is reasonable for a developer to expect the context to be rebuilt when committing a file from the IDE just as if they had edited the file via the UI.

## Screenshots
n/a

## Logs
n/a

### Version
Reproduced in 3.1.13

### OS
macos

### Browser
n/a

## Additional context
Going further, when developing, it would be great if the CMS was watching the file system to see if groovy classes, the app context, anything in the plugins folder, etc. are saved, so that the app context can be rebuilt then, as this would speed up the development cycle and reduce a flurry of commits while developing tricky features (especially in areas where you have to "develop by touch" due to docs being a little thin - like plugins).

**Comment by sumerjabri** [comment]
@jdrossl this can be a FAM in Engine or CRON in deployer. Let's discuss.

**Comment by jdrossl** [comment]
@sumerjabri This is working as expected in `3.1.14-SNAPSHOT`

```
[INFO] 2021-04-05T08:31:05,655 [studio.clockTaskExecutor-25] [site.SiteServiceImpl] | Syncing database with repository for site: editorial fromCommitId = bbb8b66197dd837183030211551ab43b3ea9aae6~ 
[INFO] 2021-04-05T08:31:05,709 [studio.clockTaskExecutor-25] [site.SiteServiceImpl] | Done syncing database with repository for site: editorial fromCommitId = bbb8b66197dd837183030211551ab43b3ea9aae6~ with a final result of: true 
[INFO] 2021-04-05T08:31:05,709 [studio.clockTaskExecutor-25] [site.SiteServiceImpl] | Last commit ID for site: editorial is bbb8b66197dd837183030211551ab43b3ea9aae6 
[INFO] 2021-04-05T08:31:05,730 [pool-12-thread-1] [] [context.SiteContextManager] | ================================================== 
[INFO] 2021-04-05T08:31:05,730 [pool-12-thread-1] [] [context.SiteContextManager] | <Rebuilding site context: editorial> 
[INFO] 2021-04-05T08:31:05,731 [pool-12-thread-1] [] [context.SiteContextManager] | ================================================== 
...
```

@tfield please note that the context rebuild is not immediate, you need to wait until Studio finds the new commit and it is processed

**Comment by sumerjabri** [comment]
FAM for Engine as a future goal of 4.x

**Comment by sumerjabri** [comment]
@tfield While possible to change the clock frequency and even individual clocks, it's not advisable as you're now fiddling around with foundational processes and schedules.

**Comment by sumerjabri** [comment]
This is now fixed per #6223
